agents/vantagePro2_agent/vantage_pro2_agent.py

```diff
@@ -0,0 +1,213 @@
+from typing import Optional
+from socs.agent.vantage_pro2.vantage_pro2 import VantagePro2
+
+# from LS240_agent
+on_rtd = os.environ.get('READTHEDOCS') == 'True'
+if not on_rtd:
+    from ocs import ocs_agent, site_config
+    from ocs.ocs_twisted import Pacemaker
+    from ocs.ocs_twisted import TimeoutLock
+
+
+class VantagePro2Agent:
+    """Agent to connect to single VantagePro2 Weather Monitor Device.
+
+    Args:
+         sample_freq (double): frequency (Hz) at which the weather monitor
+         samples data. Can not be faster than 0.5 Hz. This value is converted
+         to period (sec) for time.wait(seconds)
+    """
+    # change port argument when I figure out how to generalize it!
+    def __init__(self, agent, port="/dev/ttyUSB0", sample_freq=0.5):
+        self.active = True
+        self.agent: ocs_agent.OCSAgent = agent
+        self.log = agent.log
+        self.lock = TimeoutLock()
+
+        self.port = port
+        self.module: Optional[VantagePro2] = None
+
+        if sample_freq > 0.5:
+            self.log.warn("Sample frequency too fast! Setting to 0.5Hz")
+            sample_freq = 0.5
+        self.sample_freq = sample_freq
+
+        self.initialized = False
+        self.take_data = False
+
+        # Registers weather data feed
+        agg_params = {
+            'frame_length': 60,
+        }
+        self.agent.register_feed('weather_data',
+                                 record=True,
+                                 agg_params=agg_params)
+
+    # Task functions.
+    def init_VantagePro2_task(self, session, params=None):
+        """
+        Perform first time setup of the Weather Monitor Module.
+
+        Args:
+            params (dict): Parameters dictionary for passing parameters to
+                task.
+
+        """
+        if params is None:
+            params = {}
+
+        auto_acquire = params.get('auto_acquire', False)
+
+        if self.initialized:
+            return True, "Already Initialized Module"
+
+        with self.lock.acquire_timeout(0, job='init') as acquired:
+            if not acquired:
+                self.log.warn("Could not start init because "
+                              "{} is already running".format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            session.set_status('starting')
+
+            self.module = VantagePro2(self.port)
+            print("Initialized Vantage Pro2 module: {!s}".format(
+                self.module))
+
+        self.initialized = True
+
+        # Start data acquisition if requested
+        if auto_acquire:
+            self.agent.start('acq')
+
+        return True, 'Vantage Pro2 module initialized.'
+
+    def start_acq(self, session, params=None):
+        """
+        Method to start data acquisition process.
+
+        Args:
+            sample_freq (double):
+                Frequency at which weather data is sampled.
+                Defaults to 0.5 Hz.
+
+        """
+        time.sleep(2)
```

Comment:
Why do we need to sleep here? If a sleep is required after initialization this should go at the end of the init task.